src/main.rs

```diff
@@ -0,0 +1,33 @@
+//! Tasks defined rust_actix-basic.pdf
+//! using endpoint: https://jsonplaceholder.typicode.com/
+
+#![allow(non_snake_case)] // JSON does not in general conform to rust standards, so we subdue warnings about correct case to use `serde::Deserialize` without the compiler complaining.
+
+use tokio;
+
+extern crate actix_basic;
+
+const URL: &str = "https://jsonplaceholder.typicode.com/api/v1/users";
+
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+
+    println!("******\nGetting users from {}...", URL);
+    let users = actix_basic::get_users(URL).await?;
+    println!("******\nResponse: (Address of a user):\n {:#?}\n******", users[2].address);
+
+    println!("-------------------------------------\n");
+
+    let user_str = r#"
```

Comment:
Put this in a test (idiomatic rust has unit/int tests in the same file that it's testing).

Reply:
Idimatic rust can aslo have tests reside in the `tests/` dir, no? Should I include both types? Seems redundant?

Reply:
https://doc.rust-lang.org/book/ch11-03-test-organization.html#integration-tests

From what I can tell, unit testing is in each file's contents, where integration testing assumes we are producing a lib that is exported as a crate. So unit testing is what we are after?

Reply:
The main function is where you're binary "begins".  Remove the getting and posting of users here.  Add them to a tests module in their respective handlers.  I'll make a note of where they should go.  Basically, this file should just start the server (I think you have that elsewhere).

Reply:
To add more detail, you're creating a REST Http server.  You need create 2 routes that can be hit from the outside, that wrap the jsonplaceholder calls.  Re-read the PDF and let me know if you have questions.  The actix website has a lot of examples here: https://actix.rs/